src/sidebar/store/modules/sidebar-panels.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * This module handles the state for `SidebarPanel` components used in the app.
+ * It keeps track of the "active" `panelName` (simple string) and allows the
+ * opening, closing or toggling of panels via their `panelName`. It merely
+ * retains the `panelName` state as a string: it has no understanding nor
+ * opinions about whether a given `panelName` corresponds to one or more
+ * extant `SidebarPanel` components. Only one panel (as keyed by `panelName`)
+ * may be "active" (open) at one time.
+ */
+
+'use strict';
+
+const util = require('../util');
+
+function init() {
+  return {
+    /*
+     * The `panelName` of the currently-active sidebar panel.
+     * Only one `panelName` may be active at a time, but it is valid (though not
+     * the standard use case) for multiple `SidebarPanel` components to share
+     * the same `panelName`—`panelName` is not intended as a unique ID/key.
+     *
+     * e.g. If `activePanelName` were `foobar`, all `SidebarPanel` components
+     * with `panelName` of `foobar` would be active, and thus visible.
+     *
+     */
+    activePanelName: null,
+  };
+}
+
+const update = {
+  OPEN_SIDEBAR_PANEL: function(state, action) {
+    return { activePanelName: action.panelName };
+  },
+
+  CLOSE_SIDEBAR_PANEL: function(state, action) {
+    let activePanelName = state.activePanelName;
+    if (action.panelName === activePanelName) {
+      activePanelName = null;
+    }
+    return {
+      activePanelName,
```

Comment:
I kinda like returning `state` here and skipping the local var, but thats just my preference, Maybe just a comment saying `no-op` or something

Reply:
Comments +1

However, I want to leave the local variable in place, as this is me being explicit that I am making a _copy_ of the (primitive) reference `state.activePanelName`. I do not want to return a reference to the previous state's `activePanelName` as that feels redux-violate-y. Sound OK?